src/js/d3ViewComponent.js

```diff
@@ -114,11 +139,101 @@ https://raw.githubusercontent.com/fluid-project/chartAuthoring/master/LICENSE.tx
     };
 
     // Returns a formatted string for a numeric data value based on a supplied template
-    floe.d3ViewComponent.getTemplatedDisplayValue = function(totalValue, percentageDigits, template, d) {
+    floe.d3ViewComponent.getTemplatedDisplayValue = function (totalValue, percentageDigits, template, d) {
         var percentage = floe.chartAuthoring.percentage.calculate(d.value, totalValue);
         var percentageForTemplate = percentage !== null ? percentage.toFixed(percentageDigits) : percentage;
         var output = fluid.stringTemplate(template, {label: d.label, value: d.value, percentage: percentageForTemplate, total: totalValue});
         return output;
     };
 
+    // Given a data key used to maintain object constancy in D3, a DOM
+    // element with a unique ID and the component, updates the key's
+    // value (an array of IDs) to include that ID
+    floe.d3ViewComponent.addElementIdToDataKey = function (d3Key, idToAdd, that) {
+        var keyPath = "dataKeys." + d3Key;
```

Comment:
Even more direct would be simply

var keyPath = ["dataKeys", d3Key, idToAdd],
that.applier.change(keyPath, true)

and the corresponding "DELETE" change in the remove function

Always use hashes unless there is a fantastically good reason to use arrays (that is, that you really require ordering)